src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRazorProject.cs

```diff
@@ -41,6 +42,8 @@ public override IEnumerable<RazorProjectItem> EnumerateItems(string path)
             return EnumerateFiles(_provider.GetDirectoryContents(path), path, "");
         }
 
+        public virtual IChangeToken Watch(string pattern) => _provider.Watch(pattern);
```

Comment:
I guess we'll have to query the backing file providers in that case. Sound good?